### PR TITLE
 feat(data-table): add ability to manually resize columns

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description
Adding ability to manually resize columns in the data-table. 

### What's included?

- `[resizableColumns]` input
- Resizing columns will override the width configuration 

#### Test Steps

- [ ] `ng serve`
- [ ] Go to data-table docs
- [ ] See last demo with new `toggle` to resize columns

#### General Tests for Every PR

- [ ] `ng serve --aot` still works.
- [ ] `npm run lint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build` still works.

##### Screenshots or link to StackBlitz/Plunker

resizable-cols

### CODE REVIEW

**Comment by kyleledbetter** [comment]
Not a bad first pass! I might jump in and experiment w/ the ux a touch

**Comment by JulieMarie** [comment]
You have to go hunting a little to find the resizing bar, but can't think of a possible solution.  You wouldn't want to leave it visible, because that would make it sloppy.  Maybe make it a little wider?  ¯\_(ツ)_/¯

File: src/platform/core/data-table/_data-table-theme.scss [hunk 1]
```diff
@@ -25,9 +25,19 @@
...
+  td-data-table {
+    border-bottom: 1px solid mat-color($foreground, divider);
+  }
   .td-data-table-scrollable {
     border-top: 1px solid mat-color($foreground, divider);
   }
+  .td-data-table-column-resizer {
+    &:hover, &.td-resizing {
+      .td-data-table-column-separator {
+        border-right: 1px solid mat-color($foreground, divider);
```
Comment (JulieMarie) [inline src/platform/core/data-table/_data-table-theme.scss]: Possibly make this 3px or 5px to make it easier to see/grab?  For us old eyes?  :P

**Comment by emoralesb05** [comment]
This is the intended design to make it visible only when needed, but we can debate if it should be visitable all the time maybe just faded when resizing is available.

File: src/platform/core/data-table/data-table.component.ts [hunk 2]
```diff
@@ -779,6 +815,36 @@ export class TdDataTableComponent extends _TdDataTableMixinBase implements ICont
...
+   */
+  _handleStartColumnDrag(index: number, event: MouseEvent): void {
+    this._columnClientX = event.clientX;
+    this._resizingColumn = index;
+  }
+
+  /**
+   * Calculates new width depending on new clientX of dragger column
+   */
+  _handleColumnDrag(column: ITdDataTableColumn, index: number, event: DragEvent): void {
```
Comment (jeremysmartt) [inline src/platform/core/data-table/data-table.component.ts]: can you add some comments onto here about what all these calculations are doing?

Comment (emoralesb05) [reply src/platform/core/data-table/data-table.component.ts]: done~

File: src/platform/core/data-table/_data-table-theme.scss (hunk 1 above)
Comment (emoralesb05) [reply src/platform/core/data-table/_data-table-theme.scss]: made it 2px

**Comment by emoralesb05** [comment]
- [x] add comments to code 
- [x] made separator line 2px and opaque instead of hidden when not hovering it.

File: src/platform/core/data-table/data-table.component.html [hunk 3]
```diff
@@ -17,18 +17,29 @@
...
+        [numeric]="column.numeric"
+        [active]="(column.sortable || sortable) && column === sortByColumn"
+        [sortable]="column.sortable || (sortable && column.sortable !== false)"
+        [sortOrder]="sortOrderEnum"
+        [hidden]="column.hidden"
+        (sortChange)="handleSort(column)">
+        <span [matTooltip]="column.tooltip">{{column.label}}</span>
+        <span td-column-resizer
+              *ngIf="resizableColumns"
+              draggable="true"
```
Comment (jeremysmartt) [inline src/platform/core/data-table/data-table.component.html]: Doesn't appear to be working in Firefox

**Pull request merged**